platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemReporter.java

```diff
@@ -149,6 +139,8 @@ public void report(Collection<? extends Problem> problems) {
      */
     @Override
     public void report(Problem problem, OperationIdentifier id) {
+        String taskPath = ProblemTaskPathTracker.getTaskIdentityPath();
+        problem = taskPath == null ? problem : problem.toBuilder(additionalDataBuilderFactory).taskPathLocation(taskPath).build();
```

Comment:
🤔 Does this add the task path location as an origin location or as a context location? I thought the task path should go into the context.